source/sigv4.c

```diff
@@ -2078,12 +2058,12 @@ static void generateCredentialScope( const SigV4Parameters_t * pSigV4Params,
             /* Encode parameter value if non-empty. Query parameters can have empty values. */
             if( valueLen > 0U )
             {
-                returnStatus = encodeURI( pValue,
-                                          valueLen,
-                                          pBufCur + 1U,
-                                          &valueBytesWritten,
-                                          true,
-                                          true );
+                returnStatus = SigV4_EncodeURI( pValue,
+                                                valueLen,
+                                                pBufCur + 1U,
+                                                &valueBytesWritten,
+                                                true,
+                                                false );
```

Comment:
The unit test `test_SigV4_GenerateHTTPAuthorization_Happy_Paths` is failing because of this change [line 2066]. Can you please update the tests to match the changes in your PR?

Reply:
Thanks for reviewing and apologies for missing that.
I completely forgot about this change so I excluded that my changes could ever cause a unit test failure.
I will review this now, maybe I'll propose a way to pass false here only with the flag SIGV4_HTTP_PAYLOAD_IS_UNSIGNED as to not upset other use cases.